core/lib/page-functions.js

```diff
@@ -513,6 +513,15 @@ function truncate(string, characterLimit) {
   return Util.truncate(string, characterLimit);
 }
 
+// This is to support bundled lighthouse.
+// esbuild calls every function with a builtin `__name`, whose purpose is to store the
+// real name of the function so that esbuild can rename it to avoid collisions. There is no way to
+// disable this renaming, even if esbuild minification (and thus function name mangling) is disabled.
+// Anywhere we inject dynamically generated code at runtime for the browser to process,
+// we must manually include this function (because esbuild only does so once at the top scope
+// of the bundle, which is irrelevant for code executed in the browser).
+const esbuildFunctionNameStubString = 'var __name=(fn)=>fn;';
```

Comment:
This definitely isn't ideal :/

Do we need to include this for all dynamically generated code, or just dynamically generated code that declares a function?